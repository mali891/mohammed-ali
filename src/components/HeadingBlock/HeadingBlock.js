import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import { styles } from './HeadingBlock.styles'
import { TSHIRT_SIZES } from '../../styles/settings'

const headingTypes = ['h1', 'h2', 'h3', 'h4', 'h5']

const HeadingBlock = ({ children, className, type, size, noMargin, ...other }) => {
	const HeadingTag = type
	const dynamicClassNames = classNames([className], {
		'c-heading': true,
		'c-heading--xs': size === 'xs',
		'c-heading--sm': size === 'sm',
		'c-heading--md': size === 'md',
		'c-heading--lg': size === 'lg',
		'c-heading--xl': size === 'xl',
		'c-heading__no-margin': noMargin
	})

	return (
		<HeadingTag css={styles} className={dynamicClassNames} {...other}>
			{children}
		</HeadingTag>
	)
}

HeadingBlock.propTypes = {
	children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
	noMargin: PropTypes.bool,
	size: PropTypes.oneOf(Object.values(TSHIRT_SIZES)).isRequired,
	type: PropTypes.oneOf(headingTypes).isRequired
}

export default HeadingBlock
