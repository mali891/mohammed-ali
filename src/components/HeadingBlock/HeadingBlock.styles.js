import { css } from "@emotion/core"
import { FONT_SIZES, LINE_HEIGHT, SPACING } from "../../styles"

export const styles = css`
  font-family: "Apercu Medium", "Apercu Regular", "system-ui", "sans-serif";

  &.c-heading {
    &--xs {
      font-size: ${FONT_SIZES.XS};
      line-height: ${LINE_HEIGHT.XS};
      margin-bottom: ${SPACING.XS};
    }

    &--sm {
      font-size: ${FONT_SIZES.SM};
      line-height: ${LINE_HEIGHT.SM};
      margin-bottom: ${SPACING.SM};
    }

    &--md {
      font-size: ${FONT_SIZES.MD};
      line-height: ${LINE_HEIGHT.MD};
      margin-bottom: ${SPACING.MD};
      letter-spacing: 0.1rem;
    }

    &--lg {
      font-size: ${FONT_SIZES.LG};
      line-height: ${LINE_HEIGHT.LG};
      margin-bottom: ${SPACING.LG};
      letter-spacing: 0.2rem;
    }

    &--xl {
      font-weight: 500;
      font-size: ${FONT_SIZES.XL};
      line-height: ${LINE_HEIGHT.XL};
      margin-bottom: ${SPACING.XL};
      letter-spacing: 0.3rem;
    }
  }
`
