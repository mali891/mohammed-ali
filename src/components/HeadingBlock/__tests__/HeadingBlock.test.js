import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import HeadingBlock from '../HeadingBlock'

const defaultProps = {
  size: 'md',
  type: 'span'
}

describe('HeadingBlock', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<HeadingBlock {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given the size is lg', () => {
    const { asFragment, container } = render(<HeadingBlock {...defaultProps} size="lg" />)

    expect(container.querySelector('.c-heading--lg')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given the type is h1', () => {
    const { asFragment, container } = render(<HeadingBlock {...defaultProps} type="h1" />)

    expect(container.querySelector('h1')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders correctly given a noMargin prop', () => {
    const { asFragment, container } = render(<HeadingBlock {...defaultProps} noMargin />)

    expect(container.querySelector('.c-heading--no-margin')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
