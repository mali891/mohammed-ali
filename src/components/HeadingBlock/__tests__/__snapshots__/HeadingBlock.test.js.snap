// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingBlock renders correctly 1`] = `
<DocumentFragment>
  <span
    class="c-heading c-heading--md"
    css="[object Object]"
  />
</DocumentFragment>
`;

exports[`HeadingBlock renders correctly given a noMargin prop 1`] = `
<DocumentFragment>
  <span
    class="c-heading c-heading--md c-heading__no-margin"
    css="[object Object]"
  />
</DocumentFragment>
`;

exports[`HeadingBlock renders correctly given the size is lg 1`] = `
<DocumentFragment>
  <span
    class="c-heading c-heading--lg"
    css="[object Object]"
  />
</DocumentFragment>
`;

exports[`HeadingBlock renders correctly given the type is h1 1`] = `
<DocumentFragment>
  <h1
    class="c-heading c-heading--md"
    css="[object Object]"
  />
</DocumentFragment>
`;
